spec/rubocop/cop/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,790 @@
+  end
+
+  context 'with examples with non-expectation statements' do
+    it 'does not detect offenses' do
+      expect_no_offenses(<<-RUBY)
+        describe 'aggregations' do
+          specify do
+            something
+            is_expected.to be_cool
+          end
+          it { is_expected.to be_awesome }
+        end
+      RUBY
+      expect_no_offenses(<<-RUBY)
+        describe 'aggregations' do
+          it { is_expected.to be_awesome }
+          specify do
+            something
+            is_expected.to be_cool
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'with a leading single expectation example' do
+    offensive_source = <<-RUBY
+      describe do
+        it { is_expected.to be_positive }
+        specify do
+        ^^^^^^^^^^ Aggregate with the example above.
+          is_expected.to be_enthusiastic
+          is_expected.to be_skilled
+        end
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe do
+        specify do
+          is_expected.to be_positive
+          is_expected.to be_enthusiastic
+          is_expected.to be_skilled
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with a following single expectation example' do
+    offensive_source = <<-RUBY
+      describe do
+        specify do
+          is_expected.to be_enthusiastic
+          is_expected.to be_skilled
+        end
+        it { is_expected.to be_positive }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe do
+        specify do
+          is_expected.to be_enthusiastic
+          is_expected.to be_skilled
+          is_expected.to be_positive
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with an expectation with chaining matchers' do
+    offensive_source = <<-RUBY
+      describe do
+        specify do
+          is_expected
+            .to be_enthusiastic
+            .and be_hard_working
+        end
+        it { is_expected.to be_positive }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
+      end
+    RUBY
+
+    good_source = <<-RUBY
+      describe do
+        specify do
+          is_expected
+            .to be_enthusiastic
+            .and be_hard_working
+          is_expected.to be_positive
+        end
+      end
+    RUBY
+
+    it_behaves_like 'detects and autocorrects', offensive_source, good_source
+  end
+
+  context 'with scattered aggregateable examples' do
+    offensive_source = <<-RUBY
+      describe 'aggregations' do
+        it { is_expected.to be_first }
+        specify do
+          foo
+          is_expected.to be_foo
+        end
+        it { is_expected.to be_second }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Aggregate with the example above.
```

Comment:
The offense message is misleading here. The example above is the `specify do` block with cannot be aggregated.

Reply:
Good catch. Will add a line number instead of "above".